test/eos_unit_test_helpers.hpp

```diff
@@ -71,34 +71,65 @@ inline void array_compare(int num, X &&x, Y &&y, Z &&z, ZT &&ztrue, XN xname, YN
   }
 }
 
-inline void compare_two_eoss(const EOS &test_e, const EOS &ref_e) {
+template <typename E1, typename E2>
+inline void compare_two_eoss(const E1 &&test_e, const E2 &&ref_e) {
   // compare all individual member functions with 1 as inputs,
   // this function is meant to catch mis-implementations of
   // modifiers that can be initialized in such a way as to
   // be equivalent of an unmodified eos. Best used with analytic
   // eoss.
-  REQUIRE(isClose(test_e.TemperatureFromDensityInternalEnergy(1, 1),
-                  ref_e.TemperatureFromDensityInternalEnergy(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.InternalEnergyFromDensityTemperature(1, 1),
-                  ref_e.InternalEnergyFromDensityTemperature(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.PressureFromDensityInternalEnergy(1, 1),
-                  ref_e.PressureFromDensityInternalEnergy(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.SpecificHeatFromDensityInternalEnergy(1, 1),
-                  ref_e.SpecificHeatFromDensityInternalEnergy(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.BulkModulusFromDensityInternalEnergy(1, 1),
-                  ref_e.BulkModulusFromDensityInternalEnergy(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.GruneisenParamFromDensityInternalEnergy(1, 1),
-                  ref_e.GruneisenParamFromDensityInternalEnergy(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.PressureFromDensityTemperature(1, 1),
-                  ref_e.PressureFromDensityTemperature(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.SpecificHeatFromDensityTemperature(1, 1),
-                  ref_e.SpecificHeatFromDensityTemperature(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.BulkModulusFromDensityTemperature(1, 1),
-                  ref_e.BulkModulusFromDensityTemperature(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.GruneisenParamFromDensityTemperature(1, 1),
-                  ref_e.GruneisenParamFromDensityTemperature(1, 1), 1.e-15));
-  REQUIRE(isClose(test_e.MinimumDensity(), ref_e.MinimumDensity(), 1.e-15));
-  REQUIRE(isClose(test_e.MinimumTemperature(), ref_e.MinimumTemperature(), 1.e-15));
+  INFO("reference T: " << ref_e.TemperatureFromDensityInternalEnergy(1, 1) << " test T: "
+                       << test_e.TemperatureFromDensityInternalEnergy(1, 1));
+  CHECK(isClose(test_e.TemperatureFromDensityInternalEnergy(1, 1),
+                ref_e.TemperatureFromDensityInternalEnergy(1, 1), 1.e-15));
```

Comment:
I like this improvement as well!

Reply:
I think that's due to you! I just copy-pasted it out of `test_eos_unit`, which had more detailed reporting. :)

Reply:
Haha I forgot I wrote that. Well either way, it's a useful change!